core/server/worker/src/main/java/alluxio/worker/file/FileSystemWorkerFactory.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.file;
+
+import alluxio.Registry;
+import alluxio.worker.Worker;
+import alluxio.worker.WorkerFactory;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Factory to create a {@link FileSystemWorker} instance.
+ */
+@ThreadSafe
+public final class FileSystemWorkerFactory implements WorkerFactory {
+  private static final Logger LOG = LoggerFactory.getLogger(FileSystemWorkerFactory.class);
+
+  /**
+   * Constructs a new {@link FileSystemWorkerFactory}.
+   */
+  public FileSystemWorkerFactory() {}
+
+  @Override
+  public boolean isEnabled() {
+    return true;
+  }
+
+  @Override
+  public FileSystemWorker create(Registry<Worker> registry) {
+    if (!isEnabled()) {
```

Comment:
it isn't clear to someone reading this code why we would check this. I was confused reading the code, thinking that reflection must be used to modify the `isEnabled()` method. If you think it's important for `isEnabled` to be checked in `create`, I would prefer to implement this in an `AbstractWorkerFactory`